src/cast.c

```diff
@@ -100,8 +100,28 @@ Expression *implicitCastTo(Expression *e, Scope *sc, Type *t)
                 //printf("type %p ty %d deco %p\n", type, type->ty, type->deco);
                 //type = type->semantic(loc, sc);
                 //printf("type %s t %s\n", type->deco, t->deco);
-                e->error("cannot implicitly convert expression (%s) of type %s to %s",
-                    e->toChars(), e->type->toChars(), t->toChars());
+                char hint[128];
+                sprintf(hint, "");
+                if (t->isTypeDArray())
+                {
+                    if (TypeStruct* typeStruct = e->type->isTypeStruct())
```

Comment:
These two paths are almost the same, surely they can be merged into a single neater path that doesn't duplicate the hint message?